### PR TITLE
fix: match date -I error messages with GNU date

### LINKED ISSUE #4494: fuzzing on date: `date -I @` triggers a panic
Found with the fuzzer introduced in https://github.com/uutils/coreutils/pull/4389
```
$ ../target/debug/coreutils date -I@
thread 'main' panicked at 'Invalid format: @', src/uu/date/src/date.rs:118:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### PR DESCRIPTION
**Pull request by shanmukhateja**
Fixes #4494

### CODE REVIEW

**Comment by sylvestre** [comment]
panic isn't the right approach to manage a user facing error.

With your change:
```
$ ./target/debug/coreutils date -I@
thread 'main' panicked at 'date:` invalid argument `@` for '--iso-8601'', src/uu/date/src/date.rs:118:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
it should be like GNU:
```
date: invalid argument '@' for '--iso-8601'
Valid arguments are:
  - 'hours'
  - 'minutes'
  - 'date'
  - 'seconds'
  - 'ns'
Try 'date --help' for more information.
```

See https://github.com/uutils/coreutils/blob/main/src/uu/date/src/date.rs#L249-L253 as example 


Also, it will need a test to make sure we don't regress

**Comment by sylvestre** [comment]
@shanmukhateja are you still working on it ? :)

**Comment by shanmukhateja** [comment]
@sylvestre Yes I plan on working on it except I'm not sure how to implement SimpleError struct inside FromStr which returns Self.

In the case of panic, it halts the execution but now when I try using the struct, it complaints of trying to return Result.

Thoughts?

**Comment by sylvestre** [comment]
You should try a bit more :)
and look how it is done elsewere

**Comment by tertsdiepraam** [comment]
Have you seen my comment in the issue? You probably don't need to change this function, just make sure that clap gives an error before we even get here.

**Comment by sylvestre** [comment]
look great :)
just a missing test

File: src/uu/date/src/date.rs [hunk 1]
```diff
@@ -115,7 +115,7 @@ impl<'a> From<&'a str> for Iso8601Format {
             NS => Self::Ns,
             DATE => Self::Date,
             // Should be caught by clap
-            _ => panic!("Invalid format: {s}"),
+            _ => panic!("date:` invalid argument `{s}` for '--iso-8601'"),
```
Comment (sylvestre) [inline src/uu/date/src/date.rs:118]: maybe 
```suggestion
            _ => unreachable!(),
```
if we won't go there then

Comment (sylvestre) [reply src/uu/date/src/date.rs]: ping ?

Comment (shanmukhateja) [reply src/uu/date/src/date.rs]: Hey, this commit [`5e42b58` (#4499)](https://github.com/uutils/coreutils/pull/4499/commits/5e42b58927a19d26c5df01640e59f4c6426539e3) fixes it.

Still missing the unit test. Will ping when I have it working.

Comment (sylvestre) [reply src/uu/date/src/date.rs]: thanks

**Comment by shanmukhateja** [comment]
@sylvestre unit tests added.

**Comment by sylvestre** [comment]
please add the tests here:
https://github.com/uutils/coreutils/blob/main/tests/by-util/test_date.rs
and following their behavior :)

**Comment by shanmukhateja** [comment]
@sylvestre ..third time's the charm?

**Comment by sylvestre** [comment]
thanks :)

**Comment by shanmukhateja** [comment]
Thank you for being patient. @sylvestre

**Comment by sylvestre** [comment]
oh, don't mention it :)

**Pull request merged**